File: math3.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

console.log("Hello, What is your name?");

rl.question("", function(name) {
  console.log(`Hi ${name} thank you for getting MATH3. We hope you enjoy the new and improved MATH!!!`);

  setTimeout(function() {
    console.log("If you get the math wrong you start over!!");
    setTimeout(askQuestion, 1000);
  }, 1000);

  function askQuestion() {
    rl.question("What is 1+1? ", function(math) {
      if (parseInt(math) === 2) {
        console.log("Thats right");
        setTimeout(askQuestion2, 2000);
      } else {
        console.log("Wrong");
        process.exit(1);
      }
    });
  }

  function askQuestion2() {
    rl.question("What is 2+2? ", function(math) {
      if (parseInt(math) === 4) {
        console.log("Thats right");
        // Continue with other questions...
        rl.close();
      } else {
        console.log("Wrong");
        process.exit(1);
      }
    });
  }

});

File: math3.cpp
#include <iostream>
#include <thread>
#include <chrono>

int main() {
    std::string name;

    std::cout << "Hello, What is your name?" << std::endl;
    std::cin >> name;

    std::cout << "Hi " << name << " thank you for getting MATH3. We hope you enjoy the new and improved MATH!!!" << std::endl;
    std::this_thread::sleep_for(std::chrono::seconds(1));

    std::cout << "If you get the math wrong you start over!!" << std::endl;
    std::this_thread::sleep_for(std::chrono::seconds(1));
    std::this_thread::sleep_for(std::chrono::seconds(1));

    int math;

    std::cout << "What is 1+1" << std::endl;
    std::cin >> math;

    if (2 == math) {
        std::cout << "Thats right" << std::endl;
    } else {
        std::cout << "Wrong" << std::endl;
        return 1;
    }

    std::this_thread::sleep_for(std::chrono::seconds(2));

    std::cout << "What is 2+2" << std::endl;
    std::cin >> math;

    if (4 == math) {
        std::cout << "Thats right" << std::endl;
    } else {
        std::cout << "Wrong" << std::endl;
        return 1;
    }

    // Repeat the pattern for the remaining questions...

    std::cout << "What is 10+10" << std::endl;
    std::cin >> math;

    if (20 == math) {
        std::cout << "Thats right" << std::endl;
    } else {
        std::cout << "Wrong" << std::endl;
    }

    std::cout << "What is " << std::endl;
    std::cin >> math;

    if (20 == math) {
        std::cout << "Thats right" << std::endl;
    } else {
        std::cout << "Wrong" << std::endl;
    }

    return 0;
}
